lib/portlayer/storage/image_cache.go

```diff
@@ -113,14 +113,14 @@ func (c *NameLookupCache) CreateImageStore(ctx context.Context, storeName string
 		return nil, os.ErrExist
 	}
 
+	c.storeCacheLock.Lock()
+	defer c.storeCacheLock.Unlock()
+
 	u, err = c.DataStore.CreateImageStore(ctx, storeName)
```

Comment:
Is CreateImageStore idempotent? If not, you can still get into situation where two callers can end up calling this function (if they are both waiting on the lock). You would need to move the lock further up to the beginning of the function to avoid this.